libraries/AC_Avoidance/AP_OABendyRuler.cpp

```diff
@@ -0,0 +1,294 @@
+    const float lookahead_step2_dist = _current_lookahead * OA_BENDYRULER_LOOKAHEAD_STEP2_RATIO;
+
+    // get ground course
+    float ground_course_deg;
+    if (ground_speed_vec.length_squared() < OA_BENDYRULER_LOW_SPEED_SQUARED) {
+        // with zero ground speed use vehicle's heading
+        ground_course_deg = AP::ahrs().yaw_sensor * 0.01f;
+    } else {
+        ground_course_deg = degrees(ground_speed_vec.angle());
+    }
+
+    // check OA_BEARING_INC definition allows checking in all directions
+    static_assert(360 % OA_BENDYRULER_BEARING_INC == 0, "check 360 is a multiple of OA_BEARING_INC");
+
+    // search in 5 degree increments around the vehicle alternating left
+    // and right. For each direction check if vehicle would avoid all obstacles
+    float best_bearing = bearing_to_dest;
+    bool have_best_bearing = false;
+    float best_margin = -FLT_MAX;
+    float best_margin_bearing = best_bearing;
+
+    for (uint16_t i = 0; i < (360 / OA_BENDYRULER_BEARING_INC); i++) {
+        float bearing_delta = i * OA_BENDYRULER_BEARING_INC * 0.5f;
+        if (i & 1) {
+            // we alternate between trying to the left and right of the target
+            bearing_delta = -bearing_delta;
+        }
+        // bearing that we are probing
+        const float bearing_test = wrap_180(bearing_to_dest + bearing_delta);
+        const float bearing_change_deg = wrap_180(bearing_test - ground_course_deg);
+
+        // don't consider 180 degree turns, as we can't predict which way we will turn
+        if (fabsf(bearing_change_deg) > 170) {
+            continue;
+        }
+
+        // ToDo: add effective groundspeed calculations using airspeed
+        // ToDo: add prediction of vehicle's position change as part of turn to desired heading
+
+        // test location is projected from current location at test bearing
+        Location test_loc = current_loc;
+        test_loc.offset_bearing(bearing_test, lookahead_step1_dist);
+
+        // calculate margin from fence for this scenario
+        float margin = calc_avoidance_margin(current_loc, test_loc);
+        if (margin > best_margin) {
+            best_margin_bearing = bearing_test;
+            best_margin = margin;
+        }
+        if (margin > _margin_max) {
+            // this bearing avoids obstacles out to the lookahead_step1_dist
+            // now check if there is a clear path to the destination
+            if (!have_best_bearing) {
+                best_bearing = bearing_test;
+                have_best_bearing = true;
+            } else if (fabsf(wrap_180(ground_course_deg - bearing_test)) <
+                       fabsf(wrap_180(ground_course_deg - best_bearing))) {
+                // replace bearing with one that is closer to our current ground course
+                best_bearing = bearing_test;
+            }
+
+            // perform second stage test in three directions looking for obstacles
+            const float test_bearings[] = { 0, 45, -45 };
```

Comment:
Should be floating point constants.  And can lose the `=`.

Reply:
changed to floating point constants and removed the "=". txs